src/typeahead/typeahead.spec.ts

```diff
@@ -0,0 +1,246 @@
+import {inject, async} from '@angular/core/testing';
+import {TestComponentBuilder} from '@angular/compiler/testing';
+import {Component, DebugElement} from '@angular/core';
+import {NGB_TYPEAHEAD_DIRECTIVES, NgbTypeahead} from './typeahead';
+import {Observable} from 'rxjs/Rx';
+import 'rxjs/add/operator/map';
+import {NgbTypeaheadWindow} from './typeahead-window';
+import {By} from '@angular/platform-browser';
+import {expectResults, getWindowLinks} from './test-common';
+
+function getWindow(element): HTMLDivElement {
+  return <HTMLDivElement>element.querySelector('ngb-typeahead-window.dropdown-menu');
+}
+
+function getDebugInput(element: DebugElement): DebugElement {
+  return element.query(By.directive(NgbTypeahead));
+}
+
+function getNativeInput(element: HTMLElement): HTMLInputElement {
+  return <HTMLInputElement>element.querySelector('input');
+}
+
+function expectWindowResults(element, expectedResults: string[]) {
+  const window = getWindow(element);
+  expect(getWindow).not.toBeNull();
+  expectResults(window, expectedResults);
+}
+
+
+describe('ngb-typeahead', () => {
+
+  describe('window', () => {
+
+    it('should be closed by default', async(inject([TestComponentBuilder], (tcb) => {
+         const html = `<input type="text" [(ngModel)]="model" [ngbTypeahead]="find"/>`;
+
+         tcb.overrideTemplate(TestComponent, html).createAsync(TestComponent).then((fixture) => {
+           fixture.detectChanges();
+           const compiled = fixture.nativeElement;
+           expect(getWindow(compiled)).toBeNull();
+         });
+       })));
+
+    it('should be opened when there are results', async(inject([TestComponentBuilder], (tcb) => {
+         const html = `<input type="text" [(ngModel)]="model" [ngbTypeahead]="find"/>`;
+
+         tcb.overrideTemplate(TestComponent, html).createAsync(TestComponent).then((fixture) => {
+           fixture.detectChanges();
+           const compiled = fixture.nativeElement;
+
+           fixture.componentInstance.model = 'one';
```

Comment:
Hmm, I'm puzzled by this tests... If I'm reading it correctly a typeahead window would open if a _model_ value is changed... But it shouldn't be the case! We should only be kicking-in the whole machinery in response to user actions.

Not sure if this is a bug in the implementation or just a trick in a test, but I would like to understand it better.